utils/dockerutil.py

```diff
@@ -95,3 +102,39 @@ def find_cgroup_filename_pattern(mountpoints, container_id):
             return os.path.join('%(mountpoint)s/system/docker/%(id)s/%(file)s')
 
     raise MountException("Cannot find Docker cgroup directory. Be sure your system is supported.")
+
+
+def image_tag_extractor(co, key):
+    if "Image" in co:
+        split = co["Image"].split(":")
+        if len(split) <= key:
+            return None
+        elif len(split) > 2:
+            # if the repo is in the image name and has the form 'docker.clearbit:5000'
+            # the split will be like [repo_url, repo_port/image_name, image_tag]. Let's avoid that
+            split = [':'.join(split[:-1]), split[-1]]
+        return [split[key]]
+    if "RepoTags" in co:
+        splits = [el.split(":") for el in co["RepoTags"]]
+        tags = []
+        for split in splits:
+            if len(split) > 2:
+                split = [':'.join(split[:-1]), split[-1]]
+            if len(split) > key:
+                tags.append(split[key])
+        if len(tags) > 0:
+            return list(set(tags))
+    return None
```

Comment:
I find the method signature quite confusing.
Why don't we just drop the `key` argument and return a `(image_name, image_tags)` pair instead ? 
This would also avoid to go through the same code path multiple times when the `key` value changes.